File: app/src/main/java/com/cesarwillymc/jpmorgantest/data/sources/search/mapper/SearchResultMapperImpl.java
package com.cesarwillymc.jpmorgantest.data.sources.search.mapper;

import androidx.annotation.NonNull;

import com.cesarwillymc.jpmorgantest.data.sources.search.entities.WeatherDetailResponse;
import com.cesarwillymc.jpmorgantest.domain.usecase.entities.WeatherDetail;
import com.cesarwillymc.jpmorgantest.util.constants.ConstantsKt;

import javax.inject.Inject;

/**
 * Created by Cesar Canaza on 10/4/23.
 * cesarwilly.mc@gmail.com
 * <p>
 * IOWA, United States.
 */
public class SearchResultMapperImpl implements SearchResultMapper {
    @Inject
    public SearchResultMapperImpl(){}

    @NonNull
    @Override
    public WeatherDetail fromResponseToDomain(@NonNull WeatherDetailResponse info) {
        return new WeatherDetail(
                info.getName(),
                info.getMain().getFeelsLike(),
                info.getWeather().isEmpty() ? ConstantsKt.EMPTY_STRING : info.getWeather().get(ConstantsKt.ZERO).getDescription(),
                info.getMain().getHumidity(),
                info.getMain().getPressure(),
                info.getWind().getSpeed(),
                info.getSys().getSunrise(),
                info.getSys().getSunset(),
                info.getWeather().isEmpty() ? ConstantsKt.EMPTY_STRING : info.getWeather().get(ConstantsKt.ZERO).getIcon()
        );
    }
}

File: app/src/main/java/com/cesarwillymc/jpmorgantest/data/sources/search/mapper/SearchResultMapperImpl.kt
package com.cesarwillymc.jpmorgantest.data.sources.search.mapper

import com.cesarwillymc.jpmorgantest.data.sources.search.entities.WeatherDetailResponse
import com.cesarwillymc.jpmorgantest.domain.usecase.entities.WeatherDetail
import javax.inject.Inject

/**
 * Created by Cesar Canaza on 10/3/23.
 * cesarwilly.mc@gmail.com
 *
 * IOWA, United States.
 */
class SearchResultMapperImpl @Inject constructor() : SearchResultMapper {
    override fun fromResponseToDomain(info: WeatherDetailResponse): WeatherDetail {
        return WeatherDetail(
            cityName = info.name,
            temperature = info.main.feelsLike,
            weatherDescription = info.weather.firstOrNull()?.description.orEmpty(),
            humidity = info.main.humidity,
            pressure = info.main.pressure,
            windSpeed = info.wind.speed,
            sunrise = info.sys.sunrise,
            sunset = info.sys.sunset,
            icon = info.weather.firstOrNull()?.icon
        )
    }
}
